src/python/pants/jvm/compile.py

```diff
@@ -34,6 +34,10 @@ class ClasspathSourceAmbiguity(Exception):
     """Too many compiler instances were compatible with a CoarsenedTarget."""
 
 
+class ClasspathSourceRootOnlyWasInner(Exception):
```

Comment:
`SourceRoot` makes me think of `src/java`. Maybe `ClasspathRootOnlyWasInner`, or something more vague